meteor/client/ui/Settings/ShowStyle/BlueprintConfiguration.tsx

```diff
@@ -0,0 +1,176 @@
+import { BlueprintConfigSchemaSettings } from '../BlueprintConfigSchema'
+import { SomeObjectOverrideOp } from '@sofie-automation/corelib/dist/settings/objectWithOverrides'
+import { Blueprints, ShowStyleBases } from '../../../collections'
+import { useTranslation } from 'react-i18next'
+import { MappingsExt } from '@sofie-automation/corelib/dist/dataModel/Studio'
+import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { EditAttribute } from '../../../lib/EditAttribute'
+import { RedirectToBlueprintButton } from '../../../lib/SettingsNavigation'
+import { unprotectString } from '@sofie-automation/corelib/dist/protectedString'
+import { BlueprintId } from '@sofie-automation/corelib/dist/dataModel/Ids'
+import { ShowStyleBase, SourceLayers } from '../../../../lib/collections/ShowStyleBases'
+
+interface ShowStyleBaseBlueprintConfigurationSettingsProps {
+	showStyleBase: ShowStyleBase
+
+	schema: JSONSchema | undefined
+
+	layerMappings: { [studioId: string]: MappingsExt } | undefined
+	sourceLayers: SourceLayers | undefined
+}
+
+export function ShowStyleBaseBlueprintConfigurationSettings(
+	props: ShowStyleBaseBlueprintConfigurationSettingsProps
+): JSX.Element {
+	const { t } = useTranslation()
+
+	const translationNamespaces = useMemo(
+		() => ['blueprint_' + props.showStyleBase.blueprintId],
+		[props.showStyleBase.blueprintId]
+	)
+
+	const saveBlueprintConfigOverrides = useCallback(
+		(newOps: SomeObjectOverrideOp[]) => {
+			ShowStyleBases.update(props.showStyleBase._id, {
+				$set: {
+					'blueprintConfigWithOverrides.overrides': newOps,
+				},
+			})
+		},
+		[props.showStyleBase._id]
+	)
+
+	return (
+		<>
+			<h2 className="mhn">{t('Blueprint Configuration')}</h2>
+
+			<SelectBlueprint showStyleBase={props.showStyleBase} />
+			<SelectConfigPreset showStyleBase={props.showStyleBase} />
+
+			<BlueprintConfigSchemaSettings
+				schema={props.schema}
+				translationNamespaces={translationNamespaces}
+				layerMappings={props.layerMappings}
+				sourceLayers={props.sourceLayers}
+				configObject={props.showStyleBase.blueprintConfigWithOverrides}
+				saveOverrides={saveBlueprintConfigOverrides}
+				alternateConfig={undefined}
+			/>
+		</>
+	)
+}
+
+interface SelectBlueprintProps {
+	showStyleBase: ShowStyleBase
+}
+function SelectBlueprint({ showStyleBase }: SelectBlueprintProps) {
+	const { t } = useTranslation()
+
+	const allShowStyleBlueprints = useTracker(() => {
+		return Blueprints.find({
+			blueprintType: BlueprintManifestType.SHOWSTYLE,
+		}).fetch()
+	}, [])
+	const blueprintOptions: { name: string; value: BlueprintId | null }[] = useMemo(() => {
+		if (allShowStyleBlueprints) {
+			return allShowStyleBlueprints.map((blueprint) => {
+				return {
+					name: blueprint.name ? blueprint.name + ` (${blueprint._id})` : unprotectString(blueprint._id),
```

Comment:
Wouldn't this be cleaner?

```suggestion
					name: blueprint.name ? `${blueprint.name} (${blueprint._id})` : unprotectString(blueprint._id),
```